build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/util/AqlDependenciesHelper.java

```diff
@@ -53,14 +74,112 @@ public void setFlatDownload(boolean flat) {
         this.downloader.setFlatDownload(flat);
     }
 
-    public Set<DownloadableArtifact> collectArtifactsToDownload(String aql, boolean explode) throws IOException {
-        Set<DownloadableArtifact> downloadableArtifacts = Sets.newHashSet();
-        if (StringUtils.isNotBlank(buildName)) {
-            aql = addBuildToQuery(aql);
+    private List<AqlSearchResult.SearchEntry> filterAqlSearchResultsByBuild(List<AqlSearchResult.SearchEntry> searchResults) throws IOException {
+        Map<String,Boolean> buildArtifactsSha1 = fetchBuildArtifactsSha1();
+        return filterBuildAqlSearchResults(searchResults,buildArtifactsSha1);
+    }
+
+    /**
+     * Filter search results by the following priorities:
+     * 1st priority: Match {Sha1, build name, build number}
+     * 2nd priority: Match {Sha1, build name}
+     * 3rd priority: Match {Sha1}
+     */
+    private List<AqlSearchResult.SearchEntry> filterBuildAqlSearchResults(List<AqlSearchResult.SearchEntry> itemsToFilter, Map<String,Boolean> buildArtifactsSha1){
+        List<AqlSearchResult.SearchEntry> filteredResults = new ArrayList<>();
+        Map<String,List<AqlSearchResult.SearchEntry>> firstPriority = new HashMap<>();
+        Map<String,List<AqlSearchResult.SearchEntry>> secondPriority = new HashMap<>();
+        Map<String,List<AqlSearchResult.SearchEntry>> thirdPriority = new HashMap<>();
```

Comment:
Let's add a description for the above maps to make it easier to read.
Also, let's add a white-space right after **Map<String,**